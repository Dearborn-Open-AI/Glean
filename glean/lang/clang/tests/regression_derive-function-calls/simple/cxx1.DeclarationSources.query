query: cxx1.DeclarationSources _
transform: [normord, []]
