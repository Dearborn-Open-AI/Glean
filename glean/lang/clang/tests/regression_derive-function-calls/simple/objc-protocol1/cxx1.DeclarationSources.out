[
  "@generated",
  {
    "key": {
      "sources": [
        {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "call" } } },
                  "scope": { "global_": { } }
                }
              },
              "signature": {
                "key": {
                  "parameters": [
                    { "name": { "key": "s" }, "type": { "key": "J *" } }
                  ],
                  "returns": { "key": "void" }
                }
              },
              "source": {
                "columnBegin": 1,
                "columnEnd": 1,
                "file": { "key": "test.m" },
                "lineBegin": 16,
                "lineEnd": 19
              }
            }
          }
        }
      ],
      "target": {
        "objcContainer": {
          "key": {
            "id": { "interface_": { "key": "J" } },
            "source": {
              "columnBegin": 1,
              "columnEnd": 4,
              "file": { "key": "test.m" },
              "lineBegin": 9,
              "lineEnd": 11
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "sources": [
        {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "call" } } },
                  "scope": { "global_": { } }
                }
              },
              "signature": {
                "key": {
                  "parameters": [
                    { "name": { "key": "s" }, "type": { "key": "J *" } }
                  ],
                  "returns": { "key": "void" }
                }
              },
              "source": {
                "columnBegin": 1,
                "columnEnd": 1,
                "file": { "key": "test.m" },
                "lineBegin": 16,
                "lineEnd": 19
              }
            }
          }
        }
      ],
      "target": {
        "objcContainer": {
          "key": {
            "id": { "protocol": { "key": "Foo" } },
            "source": {
              "columnBegin": 1,
              "columnEnd": 4,
              "file": { "key": "test.m" },
              "lineBegin": 13,
              "lineEnd": 14
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "sources": [
        {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "call" } } },
                  "scope": { "global_": { } }
                }
              },
              "signature": {
                "key": {
                  "parameters": [
                    { "name": { "key": "s" }, "type": { "key": "J *" } }
                  ],
                  "returns": { "key": "void" }
                }
              },
              "source": {
                "columnBegin": 1,
                "columnEnd": 1,
                "file": { "key": "test.m" },
                "lineBegin": 16,
                "lineEnd": 19
              }
            }
          }
        }
      ],
      "target": {
        "objcMethod": {
          "key": {
            "container": { "interface_": { "key": "J" } },
            "isAccessor": false,
            "isInstance": true,
            "isOptional": false,
            "selector": { "key": [ "checkProtocol" ] },
            "signature": {
              "key": {
                "parameters": [
                  { "name": { "key": "f" }, "type": { "key": "Protocol *" } }
                ],
                "returns": { "key": "void" }
              }
            },
            "source": {
              "columnBegin": 1,
              "columnEnd": 36,
              "file": { "key": "test.m" },
              "lineBegin": 10,
              "lineEnd": 10
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "sources": [
        {
          "objcContainer": {
            "key": {
              "id": { "interface_": { "key": "J" } },
              "source": {
                "columnBegin": 1,
                "columnEnd": 4,
                "file": { "key": "test.m" },
                "lineBegin": 9,
                "lineEnd": 11
              }
            }
          }
        },
        {
          "objcMethod": {
            "key": {
              "container": { "interface_": { "key": "J" } },
              "isAccessor": false,
              "isInstance": true,
              "isOptional": false,
              "selector": { "key": [ "checkProtocol" ] },
              "signature": {
                "key": {
                  "parameters": [
                    { "name": { "key": "f" }, "type": { "key": "Protocol *" } }
                  ],
                  "returns": { "key": "void" }
                }
              },
              "source": {
                "columnBegin": 1,
                "columnEnd": 36,
                "file": { "key": "test.m" },
                "lineBegin": 10,
                "lineEnd": 10
              }
            }
          }
        }
      ],
      "target": {
        "objcContainer": {
          "key": {
            "id": { "interface_": { "key": "Protocol" } },
            "source": {
              "columnBegin": 1,
              "columnEnd": 1,
              "file": { "key": "<builtin>" },
              "lineBegin": 1,
              "lineEnd": 1
            }
          }
        }
      }
    }
  }
]